### PR TITLE
Fix multiple RelVals errors with `Unknown trigger name` exception 

### LINKED ISSUE #44895: Multiple RelVals: `Unknown trigger name` exception
RelVal 1042.0 is failing in multiple IBs with the following exception:
```
----- Begin Fatal Exception 02-May-2024 13:41:35 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Processing  Event run: 305064 lumi: 38 event: 58033402 stream: 0
   [1] Running path 'ALCARECOStreamPPSCalTrackBasedSelOutPath'
   [2] Calling OutputModule prePrefetchSelection()
Exception Message:
EventSelector::init, An OutputModule is using SelectEvents
to request a trigger name that does not exist
The unknown trigger name is: pathALCARECOPPSCalTrackBasedSel
----- End Fatal Exception -------------------------------------------------
```
[full log](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc12/CMSSW_14_1_X_2024-05-02-1100/pyRelValMatrixLogs/run/1042.0_RunExpressPhy2017F/step3_RunExpressPhy2017F.log#/19-19)

### PR DESCRIPTION
**Pull request by mmusich**
fixes #44895 

#### PR description:

Title says it all, minor technical fixes to avoid relval failures in IB.
No regression expected.

#### PR validation:

Now `runTheMatrix.py -l 1001.3,1040.1,1041.0,1042.0 -t 4 -j 8 --ibeos` runs fine.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * enable_tests = threading
   * workflows_threading = 1001.3, 1040.1, 1041.0, 1042.0

**Comment by mmusich** [comment]
type bug-fix

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by srimanob** [comment]
+Upgrade

**Comment by rappoccio** [comment]
+1

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2463,7 +2463,7 @@ def lhegensim2018ml(fragment,howMuch):
                         '--datatier':'ALCARECO',
                         '--eventcontent':'ALCARECO',
                         '--conditions':'auto:run2_data',
-                        '--triggerResultsProcess':'RECO',
+                        '--triggerResultsProcess':'ALCARECO',
```
Comment (grzanka) [inline Configuration/PyReleaseValidation/python/relval_steps.py:2466]: @mmusich can you explain, just for my understanding what is the difference here between ALCARECO and RECO ?

Comment (mmusich) [reply Configuration/PyReleaseValidation/python/relval_steps.py:2466]: @grzanka none, it's a convention, but you need to keep it in synch with the one you used in the step before:
https://github.com/cms-sw/cmssw/blob/8391245a8a5febd6fdaec1e66395b67f9d2dd3db/Configuration/PyReleaseValidation/python/relval_steps.py#L2407

Comment (grzanka) [reply Configuration/PyReleaseValidation/python/relval_steps.py:2466]: Thanks for explanation. Does it mean that following this convention makes this PR to fix the reported issue ?

**Pull request merged**